library/entropy.c

```diff
@@ -203,11 +203,26 @@ static int entropy_update( mbedtls_entropy_context *ctx, unsigned char source_id
     header[0] = source_id;
     header[1] = use_len & 0xFF;
 
+    /*
+     * Start the accumulator if this has not already happened. Note that
+     * it is sufficient to start the accumulator here only because all calls to
+     * gather entropy eventually execute this code.
+     */
 #if defined(MBEDTLS_ENTROPY_SHA512_ACCUMULATOR)
```

Comment:
I don't see a concrete problem with the current code, but I don't feel fully comfortable introducing the constraint that whenever `ctx->accumulator` is used, `entropy_update` must have been called somewhere before. 

As long as `entropy_init` is not allowed to fail, we could at least safeguard the part of `mbedtls_entropy_func` where `accumulator` is used, checking that `accumulator_started == 1` before calling `mbedtls_shaXXX_finish_ext` there. @andresag01 @yanesca What do you think?